client/src/components/MoreDetails.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useRef, useEffect } from 'react';
+import Menu from 'plaid-threads/Icons/MenuS1';
+import Dropdown from 'plaid-threads/Dropdown';
+import IconButton from 'plaid-threads/IconButton';
+import Touchable from 'plaid-threads/Touchable';
+
+import { LinkButton } from '.';
+import { useOnClickOutside } from '../hooks';
+import { useLink } from '../services';
+
+interface Props {
+  setBadStateShown: boolean;
+  handleSetBadState: () => void;
+  userId: number;
+  itemId: number;
+  handleDelete: () => void;
+}
+
+// Provides for testing of the ITEM_LOGIN_REQUIRED webhook and Link update mode
+export function MoreDetails(props: Props) {
+  const [menuShown, setmenuShown] = useState(false);
+  const [token, setToken] = useState('');
+  const refToButton = useRef<HTMLDivElement>(null);
+  const refToMenu: React.RefObject<HTMLDivElement> = useOnClickOutside({
+    callback: () => {
+      setmenuShown(false);
+    },
+    ignoreRef: refToButton,
+  });
+
+  const { generateLinkToken, linkTokens } = useLink();
+  // creates new link token for each item in bad state
+  useEffect(() => {
+    generateLinkToken(props.userId, props.itemId); // itemId is set because link is in update mode
+  }, [props.userId, props.itemId, generateLinkToken]);
+
+  useEffect(() => {
+    setToken(linkTokens.byItem[props.itemId]);
+  }, [linkTokens, props.itemId]);
+
+  // display choice, depending on whether item is in "good" or "bad" state
+  const linkChoice = props.setBadStateShown ? (
+    // handleSetBadState uses sandbox/item/reset_login to send the ITEM_LOGIN_REQUIRED webhook;
+    // app responds to this webhook by setting item to "bad" state (server/webhookHandlers/handleItemWebhook.js)
+    <Touchable className="menuOption" onClick={props.handleSetBadState}>
+      Test Item Login Required Webhook
```

Comment:
The margins around this get a little wonky I think because this text is long enough to wrap to the second line. See screenshot.
<img width="572" alt="Screen Shot 2021-07-15 at 10 01 11 AM" src="https://user-images.githubusercontent.com/86379616/125811969-20a204af-41ba-4883-908c-32ac8adba6c0.png">

Reply:
fixed